providers/coordinator.go

```diff
@@ -26,20 +26,23 @@ import (
 
 var BuiltinCoreID = coreconf.Config.ID
 
-var Coordinator = coordinator{
+var GlobalCoordinator = Coordinator{
```

Comment:
Why expose the coordinator?

Reply:
because now you can call `GlobalCoordinator.NewChild()`. That one needs to return a `Coordinator`, which isn't possible if it's not exposed. If we don't want to expose the struct, then we should define an interface and return that